src/commons/mobileWorkspace/MobileKeyboard.tsx

```diff
@@ -31,53 +33,86 @@ const MobileKeyboard: React.FC<Props> = props => {
       document.getElementById('mobile-floating-toggle')!.style.setProperty('width', '42px');
       document.getElementById('mobile-floating-toggle')!.style.setProperty('opacity', '0.6');
       setButtonContent('ᐸ');
-      setIsKeyoardShown(false);
+      setIsKeyboardShown(false);
     } else {
       //do showing
       document.getElementById('mobile-keyboard-toggle')!.style.setProperty('display', 'flex');
       document.getElementById('mobile-floating-toggle')!.style.setProperty('width', '99%');
       document.getElementById('mobile-floating-toggle')!.style.setProperty('opacity', '1');
       setButtonContent('ᐳ');
-      setIsKeyoardShown(true);
+      setIsKeyboardShown(true);
     }
   };
 
-  const handleRowToggle = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
-    setSelectedIndex((selectedIndex + 1) % 3);
-    const keyboardClass = document.getElementsByClassName('simple-keyboard-shortcut');
-    Array.from(keyboardClass as HTMLCollectionOf<HTMLElement>)[0].style.top =
-      -selectedIndex * 45 + 'px';
-  };
-
   const handleKeyPress = (button: string) => {
     if (!props.targetKeyboardInput) {
       return;
     }
-    const editor = props.targetKeyboardInput;
-    if (button === '{arrowleft}') {
+
+    setTargetKeyboardInput(props.targetKeyboardInput);
+    setLastKeyPressed(button);
+  };
+
+  const handleTouchStart = (e: React.TouchEvent<HTMLDivElement>) => {
+    // Get the touch coordinates and current time
+    const touch = e.touches[0];
+    setTouchStartInfo({
+      x: touch.clientX,
+      y: touch.clientY,
+      time: Date.now()
+    });
+  };
+
+  const handleTouchEnd = (e: React.TouchEvent<HTMLDivElement>) => {
+    // Compare the end position with the start position
+    const touch = e.changedTouches[0];
+    const deltaX = touch.clientX - touchStartInfo.x;
+    const deltaY = touch.clientY - touchStartInfo.y;
+    const deltaTime = Date.now() - touchStartInfo.time;
+
+    // Define thresholds for what you consider a swipe vs a tap
+    const swipeThreshold = 30; // pixels
+    const tapThresholdTime = 200; // milliseconds
+
+    if (Math.abs(deltaX) > swipeThreshold || Math.abs(deltaY) > swipeThreshold) {
+      console.log('Swipe detected');
+    } else if (deltaTime < tapThresholdTime) {
+      console.log('Tap detected');
+      handleKeyRelease();
+    } else {
+      // Handle other cases or ignore
+      console.log('Other touch event detected');
+    }
```

Comment:
We don't need the `console.log` statements right?

Reply:
Nope. I've removed them.